src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -136,10 +136,7 @@ case class AppDefinition(
         Some(proto.getCmd.getValue)
       else None
 
-    val argsOption =
-      if (commandOption.isEmpty)
-        Some(proto.getCmd.getArgumentsList.asScala.to[Seq])
-      else None
+    val argsOption = Some(proto.getCmd.getArgumentsList.asScala.to[Seq]).filter(_.nonEmpty)
```

Comment:
We can't merge this as-is, since it disallows the valid case where `args` is an empty list (it would always result in `argsOption` having the value `None`).